includes/rest/class-nodeinfo-controller.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * NodeInfo REST-Class file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Rest;
+
+use function Activitypub\get_total_users;
+use function Activitypub\get_active_users;
+use function Activitypub\get_rest_url_by_path;
+use function Activitypub\get_masked_wp_version;
+
+/**
+ * ActivityPub NodeInfo REST-Class.
+ *
+ * @author Matthias Pfefferle
+ *
+ * @see https://nodeinfo.diaspora.software/
+ */
+class Nodeinfo_Controller extends \WP_REST_Controller {
+	/**
+	 * The namespace of this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $namespace = ACTIVITYPUB_REST_NAMESPACE;
+
+	/**
+	 * The REST base for this controller's route.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'nodeinfo';
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => \WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+			)
+		);
+
+		\register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<version>\d\.\d)',
+			array(
+				'args' => array(
+					'version' => array(
+						'description' => 'The version of the NodeInfo schema.',
+						'type'        => 'string',
+						'enum'        => array( '2.0' ),
```

Comment:
The idea of the `activitypub_rest_nodeinfo_pre` filter is to allow other plugins to add their own NodeInfo versions/endpoints. with this limitation to 2.0 this is no longer possible.

Reply:
Fixed in 6a0fed3.